SSSD/sssd
# passkey: verify assertion

Include the assertion verification in the passkey helper process. For that purpose, iterate all the passkey devices connected to the system, prepare the assertion request data, decode the public key, get the authenticator data, get the device options and compare them with the organization policy and finally verify the assert.

Resolves: https://github.com/SSSD/sssd/issues/6228

## How to test

The words in bold should be replaced by a real value.

Generate the key registration:
$ ./passkey_child --register --username=**user** --domain=**test.com**
Enter PIN: 
Please touch the device.
passkey:jYXUcpMDflraOutYQ34tesRq7m3QSthrsuPLfNm+vPIJZVwTQf6BcQg935SouNqRevu4vVJI73inw7qI3ICFAw==,yGGMi6liZ8Y2m841WT0cI9Mfd3jE5H08hWKlRqvK3aDUn3vi27kbBKtbCnAoJri3vB/fZ4xBW+Kz2/yUYNbbyQ==

The following options can also be set when executing the previous command to print debugging logs
--debug-level=9 --logger=stderr

Authenticate the user. The output from the previous command should be used as an input for this one (key handle and public key). Remember that you'll need to touch the device, input the PIN and touch the device again, as the helper process is supposed to be called from SSSD and the prompts will be handled there.
$ ./passkey_child --authenticate --username=**user** --domain=**test.com** --key-handle=**jYXUcpMDflraOutYQ34tesRq7m3QSthrsuPLfNm+vPIJZVwTQf6BcQg935SouNqRevu4vVJI73inw7qI3ICFAw==** --public-key=**yGGMi6liZ8Y2m841WT0cI9Mfd3jE5H08hWKlRqvK3aDUn3vi27kbBKtbCnAoJri3vB/fZ4xBW+Kz2/yUYNbbyQ==**
